mm/mm_gran/mm_pgalloc.c

```diff
@@ -123,7 +123,7 @@ void mm_pginitialize(FAR void *heap_start, size_t heap_size)
 
 void mm_pgreserve(uintptr_t start, size_t size)
 {
-  gran_reserve(g_pgalloc, start, size);
+  DEBUGASSERT(gran_reserve(g_pgalloc, start, size) != NULL);
```

Comment:
Now that gran_reserve returns a pointer need to revert this test back to a debugassert

Reply:
you can't since DEBUGASSERT may become emtpy in release build. You need either removd the assertion or assign the returned value to a local var and assert the new variable instead.

Reply:
Doesn't DEBUGVERIFY also get removed in non-release builds ?

Reply:
Ah now I see what you mean:

`#  define DEBUGASSERT(f) ((void)(1 || (f)))`

This will expand into ((void) 1 || ((gran_reserve(g_pgalloc, start, size) != NULL)))

And gran_reserve does not get executed. My bad.